Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabLoader.cpp

```diff
@@ -391,6 +392,7 @@ namespace AzToolsFramework
                 return false;
             }
             m_prefabSystemComponentInterface->SetTemplateDirtyFlag(templateId, false);
+            PrefabSaveNotificationBus::Event(templateId, &PrefabSaveNotifications::OnPrefabSaved);
```

Comment:
question: I just want to check how a give system knows which `templateId` to connect to (e.g. with the View Bookmark system). Would it need to ask the Prefab system, what is the root opened prefab, and then connect to the save notification for that id? I just want to check I'm understanding the right way to use this in context, thanks!

Reply:
PrefabEditorEntityOwnershipInterface::GetRootPrefabTemplateId() will give you the template id of the currently loaded level in the editor. Be sure to call this after the level is loaded.

Reply:
Thanks for the info, is there a way to know/check when the level is loaded?

Reply:
You can listen to this notification which gets sent when the level is loaded. If you face any issues with that, reach out to me. We can discuss if a new notification can be sent from the PrefabEditorEntityOwnershipService instead.

Reply:
To listen to that notification he'd need to register to the bus, which needs the templateId as the address. So I guess you could listen to OnContextReset from the Prefab EOS notifications?

Reply:
There's AzFramework::EntityOwnershipServiceNotificationBus::Event(
            m_entityContextId, &AzFramework::EntityOwnershipServiceNotificationBus::Events::OnEntityOwnershipServiceReset); that you can listen too. Once the EOS is reset, the listening system needs to query again for the GetRootPrefabTemplateId() since that could have changed.

Reply:
That's pretty much the gist of it if you only care about the root prefab(level). If you want to expand the behavior to any prefab, the ebus added by Danilo should be expanded to also send out notifications when templates get loaded from disk.